### DISCUSSION TITLE
Documentation but in Javascript instead of typescript ?

### DISCUSSION THREAD

**Discussion by ExtReMLapin** [element-plus/element-plus#4349] [Q&A]
Hello,

Is there any plan to move back the documentation to javascript instead of typescript ?

**Comment by sxzz** [comment]
So why we should use JavaScript? I don't know of any benefits from switching to javascript.

I think TypeScript is better.

**Reply by ExtReMLapin** [reply]
Hello, you're missing my point, documentation used to be in JavaScript, exactly like Element **UI**, I am (probably other people) used to it and it might make people reluctant to use the newer documentation if it's a language they can't directly use (copy/paste->edit).

Right now, when it's the first time I use a component I have to go to the older documentation to see the JS example as the newer one doesn't allow use to switch between TS and JS.

Bear with me, I'm not saying we need to replace TypeScript with JavaScript, I'm saying we need a JavaScript version of the examples.

* Easier transition for the people not familiar with TypeScript
* Could allow people not knowing TypeScript to get familiar with is as they could compare the TS/JS version and see what changes.

**Reply by emojiiii** [reply]
> So why we should use JavaScript? I don't know of any benefits from switching to javascript.
> 
> I think TypeScript is better.

typescript  is very good, but may be have some new guy don't understand `typescript`
may be we can use `script` to generate a javascript example ?

**Reply by OneSeven** [reply]
It is true that TS should not be used, or TS should not be used only. Those who know TS will definitely know JS, but those who know JS may not know TS. Now many component libraries are JS+TS. This is a step backwards, it's entirely in accordance with personal preference ！！！

@JeremyWuuuuu @zazzaz

**Reply by sxzz** [reply]
As far as i know, there are most examples use javascript. Could you please figure out which examples that uses typescript?

**Reply by sxzz** [reply]
We just add `lang="ts"`, but you can remove it, and annotations/types if they really exists.

**Reply by ExtReMLapin** [reply]
> As far as i know, there are most examples use javascript. Could you please figure out which examples that uses typescript?

![](https://i.imgur.com/Dw6S9jA.gif)

On a more serious note, I only noticed it when i tried to go see the tree v2 code.

I'm surprised you tell me I can just remove the `script="ts"` thing

**Reply by ac32888** [reply]
Our project we are using JavaScript with Element UI (Older version) and when I try to use Element Plus with JavaScript code which was not working. is element plus only support Type Script?

**Reply by ExtReMLapin** [reply]
Bro it's been 2 years
